### PR TITLE
Execute tests as ordinary JUnit tests

### PR DESCRIPTION
**Pull request by HeikoKlare**
This PR adds required dependencies and build marker files to execute the DSLs and demo tests as ordinary JUnit tests rather than plug-in tests, reducing dependencies to Eclipse and improving performance.

To do so, it unifies the compilers used by the Reactions and Commonalities tests (taking the one from the Commonalities, as it works without an OSGI environment).

### CODE REVIEW

**Review by tsaglam — APPROVED** [review]

**Review by JanWittler — APPROVED** [review]

File: bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+public class InMemoryClassesCompiler {
+	private final Path javaSourcesFolder;
+	private Set<? extends Class<?>> compiledClasses;
+
+	public InMemoryClassesCompiler(Path javaSourcesFolder) {
+		this.javaSourcesFolder = javaSourcesFolder;
+	}
+
+	public InMemoryClassesCompiler compile() throws IOException {
+		checkState(compiledClasses == null, "classes have already be compiled");
```
Comment (JanWittler) [inline bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java:36]:
```suggestion
		checkState(compiledClasses == null, "classes have already been compiled");
```

File: bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import org.eclipse.xtext.xbase.testing.InMemoryJavaCompiler.Result;
+import org.eclipse.xtext.xbase.testing.JavaSource;
+import static java.lang.reflect.Modifier.isPublic;
+import com.google.common.base.Predicate;
+
+import static org.eclipse.xtext.xbase.lib.IterableExtensions.join;
+
+import static com.google.common.base.Preconditions.checkState;
+
+public class InMemoryClassesCompiler {
```
Comment (JanWittler) [inline bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java:27]: Could you add some documentation please? 🙄

File: bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+	public InMemoryClassesCompiler compile() throws IOException {
+		checkState(compiledClasses == null, "classes have already be compiled");
+		this.compiledClasses = compileJavaFiles(from(walk(javaSourcesFolder).collect(Collectors.toList()))
+				.filter(path -> path.toString().endsWith(".java")).transform(path -> new RelativeAndAbsolutePath(javaSourcesFolder, path)).toList());
+		return this;
+	}
+
+	private Set<? extends Class<?>> compileJavaFiles(Iterable<RelativeAndAbsolutePath> sourceFilePaths) {
+		InMemoryJavaCompiler compiler = new InMemoryJavaCompiler(getClass().getClassLoader(), JavaVersion.JAVA8);
```
Comment (JanWittler) [inline bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java:43]: Migration to Java 11 possible?

File: bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java (hunk 2 above)
Comment (HeikoKlare) [reply bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java:33]: Done in ed5f36a.

File: bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java (hunk 3 above)
Comment (HeikoKlare) [reply bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java]: Yes, done in 3b2a4e3.

File: bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+import org.eclipse.xtext.xbase.testing.JavaSource;
+import static java.lang.reflect.Modifier.isPublic;
+import com.google.common.base.Predicate;
+
+import static org.eclipse.xtext.xbase.lib.IterableExtensions.join;
+
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * Allows to compile all Java source files in a given folder at runtime and provides the compiles classes to use
```
Comment (JanWittler) [inline bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java:28]:
```suggestion
 * Allows to compile all Java source files in a given folder at runtime and provides the compiled classes to use
```

File: bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	 * @param javaSourcesFolder the directory to compile Java files in, must not be {@code null}
+	 */
+	public InMemoryClassesCompiler(Path javaSourcesFolder) {
+		this.javaSourcesFolder = javaSourcesFolder;
+	}
+
+	/**
+	 * Compiles the sources files in the directory passed to the constructor. Afterwards {@link #getCompiledClasses()} and
+	 * {@link #filterAndInstantiateClasses(Class, Predicate)} can be called.
+	 * @return {@code this}
```
Comment (JanWittler) [inline bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java:48]: Bit weird documentation

Comment (HeikoKlare) [reply bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java:48]: You mean the return value?

Comment (JanWittler) [reply bundles/tools.vitruv.dsls.testutils/src/tools/vitruv/dsls/testutils/InMemoryClassesCompiler.java:48]: Yes

**Review by JanWittler — APPROVED** [review]

**Pull request merged**